### PR TITLE
Stairs: Add helper function for textures

### PR DESCRIPTION
**Pull request by tenplus1**
Use function to tidy stair textures

### CODE REVIEW

File: mods/stairs/init.lua [hunk 1]
```diff
@@ -103,6 +94,16 @@ function stairs.register_stair(subname, recipeitem, groups, images, description,
 			end
 		end
```
Comment (sfan5) [inline mods/stairs/init.lua:95]: you can even optimize the logic a bit more:
```lua
	for i, image in ipairs(images) do
		stair_images[i] = type(image) == "string" and {name = image} or table.copy(image)
		if stair_images[i].backface_culling == nil then
			stair_images[i].backface_culling = true
		end
		if worldaligntex and stair_images[i].align_style == nil then
			stair_images[i].align_style = "world"
		end
	end
```

**Comment by tenplus1** [comment]
@sfan5 - Thanks, amended pull.  Also, what do you think about stairs automatically using node sounds ?

**Review by appgurueu — APPROVED** [review]

**Comment by sfan5** [comment]
> Also, what do you think about stairs automatically using node sounds ?

sounds useful

File: mods/stairs/init.lua [hunk 2]
```diff
@@ -69,10 +69,10 @@ local function get_node_vars(nodename)
 	local def = minetest.registered_nodes[nodename]
```
Comment (appgurueu) [inline mods/stairs/init.lua:69]: If I'm seeing this correctly, the only thing this function is ever used for is to "transplant" a few definition fields? Why not refactor it to be a `local function transplant_def_fields(nodename, to_def)`?

The code would then become something like

```lua
local def = minetest.registered_nodes[nodename]
if not def then return end
for _, field in ipairs({"light_source", "use_texture_alpha", "sunlight_propagates", "sounds"}) do
    if to_def[field] == nil then
        to_def[field] = def[field]
    end
end
```

Or alternatively, something like `register_node_based_on(nodename, base_nodename, def)`?

**Review by appgurueu — APPROVED** [review]
Looks fine as the "minimal change" for the sounds, but ideally `get_node_vars` with its 4 return values - which fulfill pretty much just the purpose of implementing JS's `?.` - is a pretty dirty function.

**Comment by tenplus1** [comment]
> Looks fine as the "minimal change" for the sounds, but ideally `get_node_vars` with its 4 return values - which fulfill pretty much just the purpose of implementing JS's `?.` - is a pretty dirty function.

This is how I handle it in Stairs Redo, seems a lot tidier too :)

**Review by appgurueu — APPROVED** [review]

**Pull request merged**